game/src/game/components/Controllers/AIController.cpp

```diff
@@ -76,72 +73,100 @@ void AIController::OnUpdate(const Timestep& delta_time)
     {
         handling_multiplier_ = kHandlingMultiplierReset;
     }
+}
+
+void AIController::OnUpdate(const Timestep& delta_time)
+{
+    glm::vec3 current_car_position = transform_->GetPosition();
+    glm::vec3 next_waypoint = path_to_follow_[next_path_index_];
+
+    UpdateCarControls(current_car_position, next_waypoint, delta_time);
+    NextWaypoint(current_car_position, next_waypoint);
+}
+
+void AIController::DrawDebugLine(glm::vec3 from, glm::vec3 to)
+{
+    render_service_->GetDebugDrawList().AddLine(
+        LineVertex(transform_->GetPosition()),
+        LineVertex(glm::vec3(path_to_follow_[next_path_index_].x,
+                             path_to_follow_[next_path_index_].y + 10,
+                             path_to_follow_[next_path_index_].z)));
+}
+
+void AIController::UpdateCarControls(glm::vec3& current_car_position,
+                                     glm::vec3& next_waypoint,
+                                     const Timestep& delta_time)
+{
+    // ----------------- put the code from here to updatecarcontrols
+    glm::vec3 current_car_right_direction = transform_->GetRightDirection();
+
+    VehicleCommand temp_command;
 
-    /**
-     * Here we have different criteria on how we want the car to move, the more
-     * advanced this is, the better the car will move on the track
-     * **/
-    float speed = vehicle_reference_->mPhysXState.physxActor.rigidBody
-                      ->getLinearVelocity()
-                      .magnitude();
-    // Log::debug("{}", speed);
-    if (speed <= 30)
+    float speed = vehicle_->GetSpeed();
+    if (speed <= 45)
     {
-        vehicle_reference_->mCommandState.throttle =
-            1.f * speed_multiplier_;  // for the everyone slow down pickup.
+        temp_command.throttle = 1.f * speed_multiplier_;
```

Comment:
super duper nitpicky but: could we use like `1.0f` for floats instead of `1.f` for consistency

Reply:
lol sure.